indico/web/templates/forms/location_widget.html

```diff
@@ -0,0 +1,548 @@
+                        justUseOptionSelected = resultCount === 0;
+                    }
+                }
+            });
+
+            venueInput.on('typeahead:click-on-option', function(e, data) {
+                reorderCategories(data.item_name);
+                /* The room needs to be cleared when the venue changes */
+                if (hiddenData.venue_id && data.venue_id != hiddenData.venue_id && !data.room_id) {
+                    clearField(roomInput);
+                }
+                hiddenData.venue_name = data.item_name;
+                hiddenData.venue_id = data.venue_id;
+                postSelectionActions(venueInput, data.item_name);
+                highlightOption(venueInput, data.item_name);
+                venueInput.attr('title', data.item_name);
+                latestSelectedOption = {field: venueInput.attr('id'), value: data.item_name};
+            })
+            .on('typeahead:click-on-custom-option', function(e, data) {
+                if (hiddenData.room_id) {
+                    clearField(roomInput);
+                }
+                delete hiddenData.venue_id;
+                hiddenData.venue_name = data.item_name;
+                postSelectionActions(venueInput, data.item_name);
+                venueInput.val(data.item_name);
+                validVenue = true;
+                justUseOptionSelected = false;
+                venueInput.trigger('typeahead:close-results-list');
+            });
+
+            roomInput.on('typeahead:click-on-custom-option', function(e, data) {
+                delete hiddenData.room_id;
+                hiddenData.room_name = data.item_name;
+                postSelectionActions(roomInput, data.item_name);
+                roomInput.val(data.item_name);
+                validRoom = true;
+                justUseOptionSelected = false;
+                roomInput.trigger('typeahead:close-results-list');
+            });
+
+            $('.i-location-input-field').on('typeahead:close-results-list', function() {
+                Typeahead["#" + this.id].hideLayout();
+            }).on('typeahead:open-results-list', function() {
+                Typeahead["#" + this.id].showLayout();
+            })
+            .on('focus.typeahead', function() {
+                /* In case the input is cleared, this ensures that the results dropdown will contain all available
+                options. */
+                var $this = $(this);
+                if (roomJustCleared && $this.is(roomInput)) {
+                    $this.trigger('input.typeahead');
+                    roomJustCleared = false;
+                }
+                if (venueJustChanged && $this.is(venueInput)) {
+                    $this.trigger('input.typeahead');
+                    venueJustChanged = false;
+                }
+                $('.ui-dialog-content').css('overflow', 'inherit');  // Make sure the results dropdown are displayed
```

Comment:
How can I make this more specific? It's the dialog that contains the location widget